far/stddlg.cpp

```diff
@@ -137,81 +151,188 @@ int GetSearchReplaceString(
-		ReplaceDlg[dlg_checkbox_reverse].Flags |= DIF_DISABLE; // DIF_HIDDEN ??
-	if (!Options.Regexp.has_value())
-		ReplaceDlg[dlg_checkbox_regex].Flags |= DIF_DISABLE; // DIF_HIDDEN ??
-	if (!Options.Fuzzy.has_value())
-		ReplaceDlg[dlg_checkbox_fuzzy].Flags |= DIF_DISABLE; // DIF_HIDDEN ??
-	if (!Options.PreserveStyle.has_value())
-		ReplaceDlg[dlg_checkbox_style].Flags |= DIF_DISABLE; // DIF_HIDDEN ??
+	// dlg_edit_search_text
+	SetFlagIf(dlg_edit_search_text, DIF_HIDDEN, HexVal);
+	SetFlagIf(dlg_edit_search_text, DIF_FOCUS, !HexVal);
+	SetStringIf(dlg_edit_search_text, Params.SearchStr, !HexVal);
+	SetHistory(dlg_edit_search_text, TextHistoryName);
+
+	// dlg_edit_search_hex
+	SetFlagIf(dlg_edit_search_hex, DIF_HIDDEN, !HexVal);
+	SetFlagIf(dlg_edit_search_hex, DIF_FOCUS, HexVal);
+	SetStringIf(dlg_edit_search_hex, Params.SearchStr, HexVal);
+	SetMaskIf(dlg_edit_search_hex, HasHex);
+
+	// dlg_label_replace
+	SetFlagIf(dlg_label_replace, DIF_HIDDEN, !Params.ReplaceMode);
+
+	// dlg_edit_replace
+	SetFlagIf(dlg_edit_replace, DIF_HIDDEN, !Params.ReplaceMode);
+	SetStringIf(dlg_edit_replace, Params.ReplaceStr, Params.ReplaceMode);
+	SetHistory(dlg_edit_replace, ReplaceHistoryName);
+
+	// dlg_checkbox_case
+	SetFlagIf(dlg_checkbox_case, DIF_DISABLE, !Params.CaseSensitive.has_value() || HexVal);
+	SetSelected(dlg_checkbox_case, Params.CaseSensitive.value_or(false));
+
+	// dlg_checkbox_words
+	SetFlagIf(dlg_checkbox_words, DIF_DISABLE, !Params.WholeWords.has_value() || HexVal || RexVal);
+	SetSelected(dlg_checkbox_words, Params.WholeWords.value_or(false));
+
+	// dlg_checkbox_reverse
+	SetFlagIf(dlg_checkbox_reverse, DIF_HIDDEN, !Params.Reverse.has_value());
+	SetSelected(dlg_checkbox_reverse, Params.Reverse.value_or(false));
+
+	// dlg_checkbox_regex
+	SetFlagIf(dlg_checkbox_regex, DIF_DISABLE, !Params.Regexp.has_value() || HexVal);
+	SetSelected(dlg_checkbox_regex, Params.Regexp.value_or(false));
+
+	// dlg_checkbox_fuzzy
+	SetFlagIf(dlg_checkbox_fuzzy, DIF_DISABLE, !Params.Fuzzy.has_value() || HexVal || RexVal);
+	SetSelected(dlg_checkbox_fuzzy, Params.Fuzzy.value_or(false));
+
+	// dlg_checkbox_style
+	SetFlagIf(dlg_checkbox_style, DIF_HIDDEN, !Params.ReplaceMode || !Params.PreserveStyle.has_value());
+	SetSelected(dlg_checkbox_style, Params.PreserveStyle.value_or(false));
+
+	// dlg_button_all
+	SetFlagIf(dlg_button_all, DIF_HIDDEN, Params.ReplaceMode || Params.HideButtonAll);
+
+	bool TextOrHexHotkeyUsed{};
 
 	const auto Handler = [&](Dialog* Dlg, intptr_t Msg, intptr_t Param1, void* Param2) -> intptr_t
 	{
-		if (Msg == DN_BTNCLICK && Picker && (Param1 == dlg_button_word || Param1 == dlg_button_selection))
+		switch (Msg)
 		{
-			// BUGBUG: #0003136: DM_INSERTTEXT or something like that
-			static_cast<DlgEdit*>(Dlg->GetAllItem()[dlg_edit_search].ObjPtr)->InsertString(Picker(Param1 == dlg_button_selection));
-			Dlg->SendMessage(DM_SETFOCUS, dlg_edit_search, nullptr);
-			return TRUE;
+			case DN_BTNCLICK:
```

Comment:
Minor: please format as in other places, e.g.
```C++
switch (...)
{
case a:
    ...
    break;

case b:
    {
        ...
    }
    break;

    ...
}
```

Reply:
Sure, Done. In fact, I was somewhat confused since I noticed different style in different places. Having `break` on the same line as closing curly brace is my invention though. :)